### PR TITLE
RPC: `createTransaction` support for `EvmDescription`

### PR DESCRIPTION
**Pull request by jowparks**
## Summary
Adds ability to use `createTransaction` rpc with EVM descriptions.
## Testing Plan
tests pass
## Documentation

Does this change require any updates to the Iron Fish Docs (ex. [the RPC API
Reference](https://ironfish.network/docs/onboarding/rpc/chain))? If yes, link a
related documentation pull request for the website.

```
[ ] Yes
```

## Breaking Change

Is this a breaking change? If yes, add notes below on why this is breaking and label it with `breaking-change-rpc` or `breaking-change-sdk`.

```
[ ] Yes
```

### CODE REVIEW

File: ironfish/src/rpc/routes/wallet/createTransaction.ts [hunk 1]
```diff
@@ -39,6 +40,19 @@ export type CreateTransactionRequest = {
...
+    gasPrice: string
+    gasLimit: string
+    to: string
+    value: string
+    data: string
+    privateIron: string
+    publicIron: string
+    v?: string
+    r?: string
+    s?: string
```
[the remark this answers is not in the document]
Comment (jowparks) [reply ironfish/src/rpc/routes/wallet/createTransaction.ts:54]: You can do either, add them ahead of time or add them at time of signing, that is why they are optional.

[the remark this answers is not in the document]
Comment (patnir) [reply ironfish/src/rpc/routes/wallet/createTransaction.ts:53]: Ah I wasn't aware of this change: https://github.com/iron-fish/ironfish/pull/5232

**Review by patnir — APPROVED** [review]

File: ironfish/src/rpc/routes/wallet/createTransaction.ts [hunk 2]
```diff
@@ -218,6 +246,28 @@ routes.register<typeof CreateTransactionRequestSchema, CreateTransactionResponse
       }
     }
 
+    if (request.data.evm) {
+      // if 0x is present, remove it
+      const evm = request.data.evm
+      const to = evm.to.startsWith('0x') ? evm.to.slice(2) : evm.to
```
Comment (hughy) [inline ironfish/src/rpc/routes/wallet/createTransaction.ts:252]: I feel like we're going to be doing this so much that it might be worth adding a util 😅

**Review by hughy — APPROVED** [review]

File: ironfish/src/rpc/routes/wallet/createTransaction.ts (hunk 2 above)
Comment (jowparks) [reply ironfish/src/rpc/routes/wallet/createTransaction.ts:252]: Lol probably true

Comment (jowparks) [reply ironfish/src/rpc/routes/wallet/createTransaction.ts]: Added

**Review by danield9tqh — APPROVED** [review]

**Pull request merged**